tests/unit/s2n_server_hello_test.c

```diff
@@ -104,8 +105,70 @@ int main(int argc, char **argv)
         EXPECT_SUCCESS(s2n_connection_free(server_conn));
     }
 
+     /* Test TLS 1.3 session id matching */
+    {
+        EXPECT_SUCCESS(s2n_enable_tls13());
+        struct s2n_config *client_config;
+        struct s2n_connection *client_conn;
+        EXPECT_NOT_NULL(client_config = s2n_config_new());
+        EXPECT_NOT_NULL(client_conn = s2n_connection_new(S2N_CLIENT));
+        EXPECT_SUCCESS(s2n_connection_set_config(client_conn, client_config));
+
+        struct s2n_stuffer *io = &client_conn->handshake.io;
+        /* protocol version */
+        EXPECT_SUCCESS(s2n_stuffer_write_uint8(io, 3));
+        EXPECT_SUCCESS(s2n_stuffer_write_uint8(io, 3));
```

Comment:
NP: instead of directly using the value `3` here maybe use a constant value called protocol_version?